vaadin-testbench-unit-shared/src/main/kotlin/com/vaadin/testbench/unit/internal/Shortcuts.kt

```diff
@@ -61,6 +65,34 @@ private class MockFilterJsonObject(val key: Key, val modifiers: Set<KeyModifier>
     }
 }
 
+/**
+ * This method is used to get the HashableKey object from the
+ * ShortcutRegistration class. This is needed because the HashableKey class
+ * is private, and we need to create an instance of it to test the
+ * generateEventModifierFilter method.
+ *
+ * @param keyModifier
+ * @return
+ * @throws ClassNotFoundException
+ * @throws InvocationTargetException
+ * @throws InstantiationException
+ * @throws IllegalAccessException
+ */
+@Throws(
+    ClassNotFoundException::class,
+    InvocationTargetException::class,
+    InstantiationException::class,
+    IllegalAccessException::class
+)
+public fun getHashableKey(keyModifier: Key): Any? {
+    val nestedClass = Class.forName(
+        "com.vaadin.flow.component.ShortcutRegistration\$HashableKey"
+    )
+    val ctor = nestedClass.declaredConstructors[0]
+    ctor.isAccessible = true
+    return ctor.newInstance(keyModifier)
+}
```

Comment:
Does this method need to be public? Or can we move it into the `MockFilterJsonObject` class or in the companion object and make it private or internal?
It seems to me it is not used elsewhere

Reply:
Yes, we can do this as well, makes sense as we only use it from that class.
Also I got these errors:
``` java.lang.ClassCastException: class com.vaadin.flow.component.Key$1 cannot be cast to class com.vaadin.flow.component.KeyModifier (com.vaadin.flow.component.Key$1 and com.vaadin.flow.component.KeyModifier are in unnamed module of loader 'app')```

Hencewhy, I changed some casting and the class variable types. Anytime I tried to get back the Key from the Keymodifier set, I got this.... 

Some kind of anonymous class/Kotlin magic is happening here that Key$1 is generated, but I would spare my nerves and would not go further down the road and I just fix this with the proper typing/class names in the methods... (I just wanted initially to have this strict as well, but we do enough magic here (reflection for private method, inner class constructor, emulating filterText, etc)...

I will commit your suggestion and just leaving as it is this change:

Reply:
The method replacement is Done ✅.
Please approve if you feel, have time and do not have any further requests, comments. :)